src/containers/course-section/resource-item/ResourceItem.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, Dispatch, SetStateAction } from 'react'
+import Box from '@mui/material/Box'
+import IconButton from '@mui/material/IconButton'
+import CloseIcon from '@mui/icons-material/Close'
+
+import IconExtensionWithTitle from '~/components/icon-extension-with-title/IconExtensionWithTitle'
+
+import { resourcesData } from '~/containers/course-section/CourseSectionContainer.constants'
+
+import { styles } from '~/containers/course-section/resource-item/ResourceItem.styles'
+
+import { ResourcesTabsEnum as ResourcesTypes, CourseResources } from '~/types'
+
+interface ResourceItemProps {
+  resource: CourseResources
+  setItemToDelete: Dispatch<SetStateAction<CourseResources | null>>
+}
+const ResourceItem: FC<ResourceItemProps> = ({ resource, setItemToDelete }) => {
+  const onDeleteResource = () => {
+    setItemToDelete(resource)
```

Comment:
commented above, maybe this component should accept function and pass resource to this function